src/mscorlib/src/System/IO/StreamReader.cs

```diff
@@ -817,7 +836,7 @@ public override Task<String> ReadLineAsync()
                 } while (i < tmpCharLen);
 
                 i = tmpCharLen - tmpCharPos;
-                if (sb == null) sb = new StringBuilder(i + 80);
+                if (sb == null) sb = AcquireStringBuilder(i + 80);
```

Comment:
Caution: You're in an async method here. Control flow will return to the caller at the await on line 823. Are you sure that it's OK to use the shared StringBuilder in this case?